### PR TITLE
upgrade gopkg.in/yaml.v2 to gopkg.in/yaml.v3

### PR DESCRIPTION
**Pull request by inteon**

### CODE REVIEW

**Comment by sagikazarmark** [comment]
Looks like upgrading breaks some of the tests.

**Comment by inteon** [comment]
@sagikazarmark the failing tests were due to indentation issues.
I updated the behavior to match the old indentation as close as possible & changed the indentation in the remaining tests.

File: internal/encoding/yaml/codec.go [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+)
 
 // Codec implements the encoding.Encoder and encoding.Decoder interfaces for YAML encoding.
 type Codec struct{}
 
 func (Codec) Encode(v interface{}) ([]byte, error) {
-	return yaml.Marshal(v)
+	var buf bytes.Buffer
+	encoder := yaml.NewEncoder(&buf)
+	encoder.SetIndent(1)
```
Comment (sagikazarmark) [inline internal/encoding/yaml/codec.go:15]: What's the default value for this? Would it make sense to use it instead?

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Can you please rebase from master?

File: internal/encoding/yaml/codec.go (hunk 1 above)
Comment (inteon) [reply internal/encoding/yaml/codec.go]: @sagikazarmark I created a new PR using these defaults & rebased on master: #1264

**Pull request closed without merging**